functorch/_src/aot_autograd.py

```diff
@@ -261,11 +525,39 @@ def joint_forward_backward(
                     allow_unused=True,
                 )
         backward_out_iter = iter(backward_out)
-        return outs, [
+        return mutated_inputs_and_outs_to_grad, [
             next(backward_out_iter) if i else None for i in inputs_needs_grads
         ]
 
-    return joint_forward_backward
+    def to_fun(t):
+        if isinstance(t, Tensor):
+            return torch._to_functional_tensor(t, mirror_autograd_meta=True)
+        else:
+            return t
+
+    def from_fun(t):
+        if not isinstance(t, Tensor) or not torch._is_functional_tensor(t):
+            return t
+        torch._sync(t)
+        return torch._from_functional_tensor(t)
+
+    def functionalized_joint(
+        primals: List[Any], tangents: List[Any]
+    ) -> Tuple[List[Any], List[Any]]:
+
+        # Wrap inputs into functional wrappers
+        f_primals, f_tangents = pytree.tree_map(to_fun, (primals, tangents))
+        torch._enable_functionalization(reapply_views=True)
+        try:
+            # Run the joint
+            outs = joint_forward_backward(f_primals, f_tangents)
```

Comment:
i assume this is just bc backwards may use inplace/mutation ops?  any other reason?